KeyedSemaphores/KeyedSemaphoresCollection.cs

```diff
@@ -335,5 +335,22 @@ public bool TryLock(TKey key, TimeSpan timeout, Action callback, CancellationTok
 
             return true;
         }
+
+        /// <summary>
+        ///     Check if keyed semaphore already has the provided unique key
+        /// </summary>
+        /// <param name="key">
+        ///     The unique key of this keyed semaphore
+        /// </param>
+        /// <returns>
+        ///     True when key are already locked
+        ///     False when key are available for lock
+        /// </returns>
+        public bool IsInUse(TKey key)
+        {
+            if (key == null) throw new ArgumentNullException(nameof(key));
+
+            return Index.ContainsKey(key);
```

Comment:
I still think this is wrong.
You probably want something like this:

`return Index.TryGetValue(key, out var keyedSemaphore) && keyedSemaphore.SemaphoreSlim.CurrentCount == 0`

You are assuming that a keyed semaphore being present in the index == the key is locked
But this is not necessarily true. This is a current implementation detail. 
If I changed the implementation to do a periodic cleanup of the index instead of immediate cleanup, this method would be broken.

Reply:
I have updated.

`keyedSemaphore.SemaphoreSlim.CurrentCount == 0`

I am really not sure what is proper behavior should be with when we have more than 1 in SemaphoreSlim

mb make sense return CurrentCount  in that case ?

Reply:
> I still think this is wrong. You probably want something like this:
> 
> `return Index.TryGetValue(key, out var keyedSemaphore) && keyedSemaphore.SemaphoreSlim.CurrentCount == 0`
> 
> You are assuming that a keyed semaphore being present in the index == the key is locked But this is not necessarily true. This is a current implementation detail. If I changed the implementation to do a periodic cleanup of the index instead of immediate cleanup, this method would be broken.

If you want to go so extreme though you may want to try and enter the monitor. There could be another thread currently decrementing from 1 to 0. Also the code is now buggy and will currently return `false` in virtually all circumstances (except rare race conditions).

Reply:
Apologies, amended my mistake, the current code works fine.